generated/nidcpower/session.py

```diff
@@ -4151,6 +4151,599 @@ def export_attribute_configuration_file(self, file_path):
+            "ready_for_pulse_trigger_event_pulse_polarity": {
+                'id': 1150103,
+                'property': self.ready_for_pulse_trigger_event_pulse_polarity,
+            },
+            "ready_for_pulse_trigger_event_pulse_width": {
+                'id': 1150104,
+                'property': self.ready_for_pulse_trigger_event_pulse_width,
+            },
+            "interlock_input_open": {
+                'id': 1150105,
+                'property': self.interlock_input_open,
+            },
+            "compliance_limit_symmetry": {
+                'id': 1150184,
+                'property': self.compliance_limit_symmetry,
+            },
+            "voltage_limit_high": {
+                'id': 1150185,
+                'property': self.voltage_limit_high,
+            },
+            "voltage_limit_low": {
+                'id': 1150186,
+                'property': self.voltage_limit_low,
+            },
+            "current_limit_high": {
+                'id': 1150187,
+                'property': self.current_limit_high,
+            },
+            "current_limit_low": {
+                'id': 1150188,
+                'property': self.current_limit_low,
+            },
+            "pulse_voltage_limit_high": {
+                'id': 1150189,
+                'property': self.pulse_voltage_limit_high,
+            },
+            "pulse_voltage_limit_low": {
+                'id': 1150190,
+                'property': self.pulse_voltage_limit_low,
+            },
+            "pulse_bias_voltage_limit_high": {
+                'id': 1150191,
+                'property': self.pulse_bias_voltage_limit_high,
+            },
+            "pulse_bias_voltage_limit_low": {
+                'id': 1150192,
+                'property': self.pulse_bias_voltage_limit_low,
+            },
+            "pulse_current_limit_high": {
+                'id': 1150193,
+                'property': self.pulse_current_limit_high,
+            },
+            "pulse_current_limit_low": {
+                'id': 1150194,
+                'property': self.pulse_current_limit_low,
+            },
+            "pulse_bias_current_limit_high": {
+                'id': 1150195,
+                'property': self.pulse_bias_current_limit_high,
+            },
+            "pulse_bias_current_limit_low": {
+                'id': 1150196,
+                'property': self.pulse_bias_current_limit_low,
+            },
+            "voltage_level": {
+                'id': 1250001,
+                'property': self.voltage_level,
+            },
+            "ovp_enabled": {
+                'id': 1250002,
+                'property': self.ovp_enabled,
+            },
+            "ovp_limit": {
+                'id': 1250003,
+                'property': self.ovp_limit,
+            },
+            "current_limit": {
+                'id': 1250005,
+                'property': self.current_limit,
+            },
+            "output_enabled": {
```

Comment:
I think it's unfortunate that we are duplicating all this information we already have within the Session class just in dictionary form. I understand that since we use a code-generator, it doesn't add any maintenance burden. But it just seems _wrong_. We should be looking up the member of the class Session by the given name, then trying to get the attribute ID. If that fails, we can raise the appropriate exception.

Reply:
I am trying to keep everything isolated to this method. To do what you suggest would require modifying the attributes to allow getting the attribute ID without sneaking in and reading a 'private' variable.

Reply:
>  To do what you suggest would require modifying the attributes to allow getting the attribute ID without sneaking in and reading a 'private' variable.

I think that's acceptable. It's all within the same module.

Reply:
Same module but different classes.

I will see what I can do so we can see what that would look like.